pages/entropy/generate-random-numbers/evm.mdx

```diff
@@ -14,81 +14,76 @@ npm install @pythnetwork/entropy-sdk-solidity
+Entropy also requires selecting a randomness provider. The randomness provider is a third-party
+who participates in the generation process. Each provider is identified by an address and hosts
+a keeper service for fullfilling requests.
 
 The simplest way to choose a provider is to use the default provider.
 The default provider for each contract and their corresponding URI is also listed in the
 [Entropy contract addresses](../contract-addresses).
 
-You can also get the default provider's address by calling the `getDefaultProvider` method, then
-get the provider's Fortuna URI by calling `getProviderInfo`:
+You can also get the default provider's address by calling the `getDefaultProvider` method:
 
 ```solidity copy
 address provider = entropy.getDefaultProvider();
-bytes providerUri = entropy.getProviderInfo().uri;
 ```
 
 ## Usage
 
 To generate a random number, follow these steps.
 
-### 1. Commit to a random number
+### 1. Generate a random number
 
-Generate a 32-byte random number on the client side, then hash it with keccak256 to create a commitment.
+Generate a 32-byte random number on the client side.
 You can do this with typescript and web3.js as follows:
 
 ```typescript copy
 const randomNumber = web3.utils.randomHex(32);
-const commitment = web3.utils.keccak256(randomNumber);
 ```
 
 ### 2. Request a number from Entropy
 
-Invoke the `request` method of the `IEntropy` contract.
-The `request` method requires paying a fee in native gas tokens which is configured per-provider.
-Use the `getFee` method to calculate the fee and send it as the value of the `request` call:
+Invoke the `requestWithCallback` method of the `IEntropy` contract.
+The `requestWithCallback` method requires paying a fee in native gas tokens which is configured per-provider.
+Use the `getFee` method to calculate the fee and send it as the value of the `requestWithCallback` call:
 
 ```solidity copy
 uint fee = entropy.getFee(provider);
-uint64 sequenceNumber = entropy.request{value: fee}(provider, commitment, true);
+uint64 sequenceNumber = entropy.requestWithCallback{value: fee}(provider, randomNumber);
 ```
 
-This method returns a sequence number. Store this sequence number for use in later steps.
-If you are invoking this off-chain, the method also emits a `PythRandomEvents.Requested` event that contains the sequence number in it.
+This method returns a sequence number. You can store this sequence number to identify the request in next step.
 
-### 3. Fetch the provider's number
-
-Submit a query to the `/revelations/` endpoint of the provider's Fortuna webservice.
-This endpoint takes a single path parameter, which is the sequence number of the request.
-
-```typescript copy
-await axios.get(`${providerUri}/revelations/${sequenceNumber}`);
-```
-
-This method returns a JSON object containing the provider's random number.
-
-### 4. Reveal the number
-
-Invoke the `reveal` method on the `IEntropy` contract:
+### 4. Implement callback for Entropy
```

Comment:
Some more comments here is useful:

When the final random number is ready to use, the `entropyCallback` function will be called by the `Entropy` contract. This will happen in a separate transaction submitted by the requested provider. The `entropyCallback` function should be implemented in the same contract that is requesting the random number.